To: vim_dev@googlegroups.com
Subject: Patch 7.3.453
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.453
Problem:    Pasting in the command line is slow.
Solution:   Don't redraw if there is another character to read. (Dominique
	    Pelle)
Files:	    src/ex_getln.c


*** ../vim-7.3.452/src/ex_getln.c	2012-02-22 14:58:24.000000000 +0100
--- src/ex_getln.c	2012-02-22 17:53:32.000000000 +0100
***************
*** 1852,1859 ****
  # endif
  		)
  	    /* Always redraw the whole command line to fix shaping and
! 	     * right-left typing.  Not efficient, but it works. */
! 	    redrawcmd();
  #endif
      }
  
--- 1852,1862 ----
  # endif
  		)
  	    /* Always redraw the whole command line to fix shaping and
! 	     * right-left typing.  Not efficient, but it works.
! 	     * Do it only when there are no characters left to read
! 	     * to avoid useless intermediate redraws. */
! 	    if (vpeekc() == NUL)
! 		redrawcmd();
  #endif
      }
  
*** ../vim-7.3.452/src/version.c	2012-02-22 17:37:55.000000000 +0100
--- src/version.c	2012-02-22 17:56:59.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     453,
  /**/

-- 
From "know your smileys":
 <<<:-{    Worf (Never smiles anyways, so he's a bad smiley)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
